__tests__/integration/integration.test.ts

```diff
@@ -0,0 +1,246 @@
+import { OctopusCliWrapper } from '../../src/octopus-cli-wrapper'
+// we use the Octopus API client to setup and teardown integration test data, it doesn't form part of create-release-action at this point
+import { PackageRequirement, RunCondition, StartTrigger } from '@octopusdeploy/message-contracts'
+import { Client, ClientConfiguration, Repository } from '@octopusdeploy/api-client'
+import { randomBytes } from 'crypto'
+import { CleanupHelper } from './cleanup-helper.test'
+import { RunConditionForAction } from '@octopusdeploy/message-contracts/dist/runConditionForAction'
+import { setOutput } from '@actions/core'
+
+// NOTE: These tests assume Octopus is running and connectable.
+// In the build pipeline they are run as part of a build.yml file which populates
+// OCTOPUS_TEST_URL and OCTOPUS_TEST_APIKEY environment variables pointing to docker
+// containers that are also running. AND it assumes that 'octo' is in your PATH
+//
+// If you want to run these locally outside the build pipeline, you need to launch
+// octopus yourself, and set OCTOPUS_TEST_CLI_PATH, OCTOPUS_TEST_URL and OCTOPUS_TEST_APIKEY appropriately,
+// and put octo in your path somewhere.
+// all resources created by this script have a GUID in
+// their name so we they don't clash with prior test runs
+
+const octoExecutable = process.env.OCTOPUS_TEST_CLI_PATH || 'octo' // if 'octo' isn't in your system path, you can override it for tests here
+
+const apiClientConfig: ClientConfiguration = {
+  apiKey: process.env.OCTOPUS_TEST_APIKEY || 'API-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
+  apiUri: process.env.OCTOPUS_TEST_URL || 'http://localhost:8050'
+}
+
+// experimental. Should probably be a custom jest matcher
+function expectMatchAll(actual: string[], expected: (string | RegExp)[]) {
+  expect(actual.length).toEqual(expected.length)
+  for (let i = 0; i < actual.length; i++) {
+    const a = actual[i]
+    const e = expected[i]
+    if (e instanceof RegExp) {
+      expect(a).toMatch(e)
+    } else {
+      expect(a).toEqual(e)
+    }
+  }
+}
+
+describe('integration tests', () => {
+  const runId = randomBytes(16).toString('hex')
+  const globalCleanup = new CleanupHelper()
+
+  const localProjectName = `project${runId}`
+  const standardInputParameters = makeInputParameters({
+    project: localProjectName,
+    apiKey: apiClientConfig.apiKey,
+    server: apiClientConfig.apiUri
+  })
+
+  let apiClient: Client
+  beforeAll(async () => {
+    apiClient = await Client.create(apiClientConfig)
+
+    const repository = new Repository(apiClient)
+
+    // pre-reqs: We need a project, which needs to have a deployment process
+
+    const lifeCycles = await repository.lifecycles.all()
+    const lifeCycle = lifeCycles.length > 0 ? lifeCycles[0] : undefined
+    if (!lifeCycle) throw new Error("Can't find first lifecycle")
+
+    const projectGroups = await repository.projectGroups.all()
+    const projectGroup = projectGroups.length > 0 ? projectGroups[0] : undefined
```

Comment:
Not that it merits a change but I believe in this case `const projectGroup = projectGroups[0]` will result in the same result. This is because accessing a 0 index on an empty array will return undefined.

Reply:
Oh, good to know